tests/unit/torch/test_ranking_metrics.py

```diff
@@ -64,6 +64,58 @@ def test_mean_recipricol_rank():
     )
 
 
+def test_recall_at():
+    metric = RecallAt([1, 2, 3, 4], labels_onehot=False)
+    result = metric(
+        torch.tensor(
+            [[1, 2, 3, 4, 5, 4, 3, 2, 1], [1, 2, 3, 4, 5, 4, 3, 2, 1], [1, 2, 3, 4, 5, 4, 3, 2, 1]]
+        ),
+        torch.tensor(
+            [[0, 0, 0, 0, 0, 0, 0, 1, 0], [0, 0, 0, 0, 0, 1, 0, 0, 0], [0, 0, 0, 0, 1, 0, 0, 0, 0]]
+        ),
+    )
+    assert torch.all(
+        torch.lt(
+            torch.abs(torch.add(result, -torch.tensor([0.3333, 0.3333, 0.6667, 0.6667]))), 1e-3
+        )
+    )
+
+
+def test_recall_at_3d():
+    metric = RecallAt([1, 2, 3, 4], labels_onehot=False)
+    result = metric(
+        torch.tensor(
+            [
+                [[1, 2, 3, 4, 5, 4, 3, 2, 1], [1, 2, 3, 4, 5, 4, 3, 2, 1]],
+                [[1, 2, 3, 4, 5, 4, 3, 2, 1], [1, 2, 3, 4, 5, 4, 3, 2, 1]],
+            ]
+        ),
+        torch.tensor(
+            [
+                [[0, 0, 0, 0, 0, 0, 0, 1, 0], [0, 0, 0, 0, 0, 1, 0, 0, 0]],
+                [[0, 0, 0, 0, 1, 0, 0, 0, 0], [0, 0, 0, 0, 0, 1, 0, 0, 0]],
+            ]
+        ),
+    )
+    assert torch.all(
+        torch.lt(torch.abs(torch.add(result, -torch.tensor([0.25, 0.25, 0.75, 0.75]))), 1e-3)
+    )
+
+
+@pytest.mark.parametrize("cutoff", [4, [4]])
+def test_recall_at_single_metric(cutoff):
```

Comment:
I am still getting the `TypeError: iteration over a 0-d tensor`  if I set the metrics as a single value as below: 
```
# Define the evaluation top-N metrics and the cut-offs
metrics = [NDCGAt(top_ks=[5], labels_onehot=True),  
           RecallAt(top_ks=[5], labels_onehot=True)]
```

You can run getting started notebooks 1 and 2 to repro the issue.

Reply:
Thanks for the reproducer @rnyak. I fixed it and added a test to cover this use-case.
Regarding `labels_onehot`, if it is `True` the labels will be converted to one-hot representation. Setting it to `False` should fail in this case, unless you provide the labels already one-hot encoded.